sdk/core/azure-core/src/main/java/com/azure/core/util/Header.java

```diff
@@ -25,7 +35,21 @@
     public Header(String name, String value) {
         Objects.requireNonNull(name, "'name' cannot be null.");
         this.name = name;
-        this.value = value;
+        this.values = new LinkedList<>();
+        this.values.add(value);
+    }
+
+    public Header(String name, String... values) {
+        Objects.requireNonNull(name, "'name' cannot be null.");
+        this.name = name;
+        this.values = new LinkedList<>();
+        this.values.addAll(Arrays.asList(values));
+    }
+
+    public Header(String name, List<String> values) {
+        Objects.requireNonNull(name, "'name' cannot be null.");
```

Comment:
Should we prevent `null` values here as this would cause downstream issues

Reply:
I'm ok with stripping out null values if you think this is to the spec?